main_test.go

```diff
@@ -1484,6 +1487,44 @@ func TestEntrypointWithResponseFiltering(t *testing.T) {
 	})
 }
 
+func TestIntegrationWithAuditTrail(t *testing.T) {
+	log := getLogger(t)
+
+	gockHost := setupGockServer(t, 3099, GockOptions{
+		OASTestFilePath: "./mocks/mockForResponseFilteringOnResponse.json",
+		APIs: map[string]GockReply{
+			"/filters-with-audit-data/": {
+				StatusCode: 200,
+				Response:   map[string]interface{}{"D": true},
+			},
+		},
+	})
+
+	envs := getEnvs(t, map[string]string{
+		"HTTP_PORT":               "3041",
+		"TARGET_SERVICE_HOST":     gockHost,
+		"TARGET_SERVICE_OAS_PATH": "/documentation/json",
+		"OPA_MODULES_DIRECTORY":   "./mocks/rego-policies",
+		"LOG_LEVEL":               "trace",
+		"ENABLE_AUDIT_TRAIL":      "true",
+	})
+
+	app, err := setupService(envs, log)
+	require.NoError(t, err)
+	require.True(t, <-app.sdkBootState.IsReadyChan())
+
+	t.Run("200 - with both request and response policy flow", func(t *testing.T) {
+		w := httptest.NewRecorder()
+		req := httptest.NewRequest(http.MethodGet, "http://localhost:3041/filters-with-audit-data/", nil)
+
+		app.router.ServeHTTP(w, req)
+
+		respBody, _ := io.ReadAll(w.Result().Body)
+		require.Equal(t, http.StatusOK, w.Result().StatusCode)
+		require.Equal(t, `{"D":true}`, string(respBody))
```

Comment:
is testing audit logging feature any valuable in this integration test?

Reply:
I've actually put this test up to make sure everything worked e2e and to run the debugger on the whole flow while using a policy that invokes the new builtin function.

The test was useful indeed, I've added assertions on the produced audit and it failed miserably, so it actually made me fix a bug :D